### ISSUE TITLE
[Bug]: external WMS/WFS with authentication not working

### ISSUE THREAD

**Issue by josemvm** [3liz/lizmap-web-client#3884]
### What is the bug?

i'm trying to get layers from WMS/WFS servers via https and using authentication
on qgis desktop i'm using basic authentication for both cases
for wfs layer i can add virtual fields to a wfs layer and everything works fine!

### Steps to reproduce the issue

try to add wms/wfs layer with authentication

### Versions

**Versions :**

* Lizmap Web Client : 3.6.5
* Lizmap plugin : 3.17.0
* QGIS Desktop : 3.22.16
* QGIS Server : 3.22.16
* Py-QGIS-Server : not used
* QGIS Server plugin atlasprint : 3.3.2
* QGIS Server plugin lizmap_server : 2.8.0
* QGIS Server plugin wfsOutputExtension : 1.8.0

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

### QGIS server version, only if the section above doesn't mention the QGIS Server version

3.22.16

### Operating system

Ubuntu 20.04

### Browsers

Firefox

### Browsers version

Firefox 117.0.1

### Relevant log output

```shell
An error occurred while loading this map. Some necessary resource may temporarily be unavailable. Please try again later.
Go back to the home page.

Error in project xx/xxx: GetCapabilities result code: 500
```

**Comment by Gustry** [comment]
How do you store your authentification ? Did you use the QGIS auth DB ?
What are your logs in QGIS server and/or in LWC about this 500 error ?

**Comment by josemvm** [comment]
>How do you store your authentification ? Did you use the QGIS auth DB ?

yes, i'm using basic authentication, credentials stored as plain text in project fie

**Comment by josemvm** [comment]
>What are your logs in QGIS server and/or in LWC about this 500 error ?

lizmap logs:
2023-09-22 13:55:17	10.27.10.58	lizmapadmin	Error in project ma/test_wms_auth: GetCapabilities result code: 500
2023-09-22 13:55:17	10.27.10.58	lizmapadmin	Error in project ma/test_wms_auth: GetCapabilities result code: 500
2023-09-22 13:55:17	10.27.10.58	lizmapadmin	GetCapabilities result code: 500

qgis server logs:
12:57:07 WARNING WMS[505050]: Download of capabilities failed: SSL handshake failed
12:57:07 CRITICAL Server[505050]: Error, Layer(s) mv_0702_prd_1957b0d6_df38_4c9b_9d60_b66d78c12a28 not valid in project /var/www/html/lizmap_web_client/lizmap/install/ma/test_wms_auth.qgs
12:57:07 WARNING Server[505050]: <?xml version="1.0" encoding="UTF-8"?>
<ServerException>Layer(s) not valid</ServerException>

**Comment by Gustry** [comment]
>  12:57:07 WARNING WMS[505050]: Download of capabilities failed: SSL handshake failed

It's an error with your configuration in QGIS server, not Lizmap as you can see.
QGIS server can not connect the external server with SSL handshake.

* Check that QGIS server can write in its HOME directory, to write the file about SSL

Something to check (it depends of your installation of course) `find /srv/qgis -type f -name "qgis-auth.db" -print0 | xargs -0 ls -lh`

I'm closing as there is first the issue with QGIS server it self.

**Comment by josemvm** [comment]
this my folder, files end rights on root@gxxx:/srv/qgis/.local/share/QGIS/QGIS3/profiles/default#
(having rights to write in qgis-auth.db)

20198 drwxr-xr-x 6 www-data www-data   4096 Sep 22 14:45 .
20197 drwxr-xr-x 3 www-data www-data   4096 Feb 22  2022 ..
20201 -rw-r--r-- 1 www-data www-data     13 Feb 22  2022 bookmarks.xml
20226 drwxr-xr-x 4 www-data www-data   4096 Feb 22  2022 cache
20245 drwxr-xr-x 2 www-data www-data   4096 Feb 22  2022 gdal_pam
20200 drwxr-xr-x 2 www-data www-data   4096 Feb 22  2022 palettes
20204 drwxr-xr-x 2 www-data www-data   4096 Feb 22  2022 QGIS
20246 -rw-r--r-- 1 www-data www-data  45056 Feb 22  2022 qgis-auth.db
20199 -rw-r--r-- 1 www-data www-data  23552 Jun 27  2022 qgis.db
20202 -rw-r--r-- 1 www-data www-data 495616 Apr  4  2022 symbology-style.db

**Comment by Gustry** [comment]
All files are from February 2022 ....
Search this common issue about SSL handshake in QGIS server.

We found sometimes is about writing the `qgis-auth.db` file.
